integration_tests/test_suites/daemon-test-suite/auto_run_reexecution_tests/test_auto_run_reexecution.py

```diff
@@ -422,6 +425,226 @@ def test_consume_new_runs_for_automatic_reexecution(instance, workspace_context)
     assert second_retry.tags.get(AUTO_RETRY_RUN_ID_TAG) is None
 
 
+def test_consume_new_runs_for_automatic_reexecution_mimic_daemon_fails_before_run_is_launched(
+    instance, workspace_context
+):
+    instance.wipe()
+    instance.run_coordinator.queue().clear()
+    list(
+        consume_new_runs_for_automatic_reexecution(
+            workspace_context,
+            instance.get_run_records(filters=RunsFilter(statuses=[DagsterRunStatus.FAILURE])),
+        )
+    )
+
+    assert len(instance.run_coordinator.queue()) == 0
+
+    run = create_run(instance, status=DagsterRunStatus.STARTED, tags={MAX_RETRIES_TAG: "2"})
+    dagster_event = DagsterEvent(
+        event_type_value=DagsterEventType.PIPELINE_FAILURE.value,
+        job_name="foo",
+        message="",
+    )
+    event_record = EventLogEntry(
+        user_message="",
+        level=logging.ERROR,
+        job_name="foo",
+        run_id=run.run_id,
+        error_info=None,
+        timestamp=time.time(),
+        dagster_event=dagster_event,
+    )
+    instance.handle_new_event(event_record)
+    run = instance.get_run_by_id(run.run_id)
+    assert run
+    assert run.tags.get(WILL_RETRY_TAG) == "true"
+
+    list(
+        consume_new_runs_for_automatic_reexecution(
+            workspace_context,
+            instance.get_run_records(filters=RunsFilter(statuses=[DagsterRunStatus.FAILURE])),
+        )
+    )
+    assert len(instance.run_coordinator.queue()) == 1
+    first_retry_run_id = instance.run_coordinator.queue()[0].run_id
+
+    # if the daemon fails between creating the run and submitting the run, the run will exist in the
+    # db but not the run_coordinator queue
+    instance.run_coordinator.queue().clear()
+    assert len(instance.run_coordinator.queue()) == 0
+    assert len(instance.get_runs()) == 2  # original run and retry run are both in the db
+
+    list(
+        consume_new_runs_for_automatic_reexecution(
+            workspace_context,
+            instance.get_run_records(filters=RunsFilter(statuses=[DagsterRunStatus.FAILURE])),
+        )
+    )
+
+    # because the retried run already exists, it is in the run group for the original run. Because
+    # it is in the run group of the original run, we will see that the run group contains a run with
+    # parent_run_id as the original run. Therefore the original run will not be retried
+    # This is a problem, since the retried run will be stuck in a NOT_STARTED state
```

Comment:
This edge case exists in master right? Curious if your subsequent changes are going to detect this case and re-enqueue it

Reply:
yeah, the current behavior in master is that the run won't get resubmitted. The stacked PR can detect this case and resubmit